x-pack/solutions/observability/plugins/synthetics/server/tasks/sync_private_locations_monitors_task.ts

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { TaskManagerSetupContract } from '@kbn/task-manager-plugin/server/plugin';
+import {
+  SavedObjectsClientContract,
+  SavedObjectsFindResult,
+} from '@kbn/core-saved-objects-api-server';
+import { EncryptedSavedObjectsPluginStart } from '@kbn/encrypted-saved-objects-plugin/server';
+import { ALL_SPACES_ID } from '@kbn/spaces-plugin/common/constants';
+import { RunContext } from '@kbn/task-manager-plugin/server';
+import moment from 'moment';
+import { syntheticsParamType } from '../../common/types/saved_objects';
+import { normalizeSecrets } from '../synthetics_service/utils';
+import type { PrivateLocationAttributes } from '../runtime_types/private_locations';
+import {
+  HeartbeatConfig,
+  MonitorFields,
+  SyntheticsMonitorWithSecretsAttributes,
+} from '../../common/runtime_types';
+import { MonitorConfigRepository } from '../services/monitor_config_repository';
+import { SyntheticsMonitorClient } from '../synthetics_service/synthetics_monitor/synthetics_monitor_client';
+import { getPrivateLocations } from '../synthetics_service/get_private_locations';
+import { SyntheticsServerSetup } from '../types';
+import {
+  formatHeartbeatRequest,
+  mixParamsWithGlobalParams,
+} from '../synthetics_service/formatters/public_formatters/format_configs';
+
+const TASK_TYPE = 'Synthetics:Sync-Private-Location-Monitors';
+const VERSION = '1.0.0';
```

Comment:
I think I'd advise against putting the version number in the task id.  Someone may think they can change it (for whatever reason), but then that will just add another task, assuming `1.0.0` got registered.  So that particular task would presumably be changed to not run - and I'd guess you'd have to parse the task id to get the version to check if it's the one that shouldn't be running, and should actually be deleted.  In any case, seems like a potential source of errors, if a new version is introduced.

Reply:
Done